src/Loop.php

```diff
@@ -42,12 +42,11 @@ public static function setFactory(DriverFactory $factory = null)
         }
 
         self::$factory = $factory;
+        self::$driver = null; // reset it here, it will be actually instantiated inside execute() or get()
 
         if ($factory === null) {
-            self::$driver = null;
             self::$registry = null;
         } else {
-            self::$driver = self::createDriver();
             self::$registry = [];
```

Comment:
We have an issue there anyway, because the registry isn't reused for wait currently. It's always a fresh one.

https://github.com/async-interop/event-loop/blob/2c22efbb8a4b01fdec3928842397fb1bd0a842ea/src/Loop.php#L62-L80